### ISSUE TITLE
How can I exclude the Home Page from being cached?

### ISSUE THREAD

**Issue by renzms** [websharks/comet-cache-kb#125]
---
##### KB Article Creation Checklist

- [x] Write initial draft for this KB Article; label this issue `draft` and either `questions` or `tutorials`
- [x] Add [required YAML configuration](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration)
- [x] Add Tags for this KB Article to the YAML config (see [YAML Keys (Explained)](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration#yaml-keys-explained))
- [ ] Edit and finalize draft for publishing (remove `draft` label, add `draft-finalized` label)
- [ ] Assign Issue to yourself and [create Markdown file](https://github.com/websharks/team/wiki/KB-Articles-::-Creating-the-Markdown-File) (remove `draft-finalized` label, add `pending`)
- [ ] Project Lead: Review and Publish KB Article (remove `pending` label, add `published` label)

##### Additional TODOs

_Additional TODO list items go here._

```yaml
title: How can I exclude the Home Page from being cached?
categories: questions
tags: http-referrer-exclusion-patterns, uri-exclusion-patterns, user-agent-exclusion-patterns
author: renzms
github-issue: https://github.com/websharks/comet-cache-kb/issues/125
```
---

When using Comet Cache Pro, you can use the URI Exclusion Patterns feature (see **Comet Cache → Plugin Options → URI Exclusion Patterns**) to exclude specific pages on your site from being cached. In some cases, you may only want to exclude the home page.

Using [Watered-Down Regex Syntax](https://cometcache.com/kb-article/watered-down-regex-syntax/), it is possible to set an exclusion that will match at the beginning or end of the URI. The solution here is to use the `^` and `$` chars in the URI Exclusion Patterns panel. 

```
^/exact-match/$
```

In the URI Exclusion Patterns panel (see **Dashboard → Comet Cache → Plugin Options → URI Exclusion Patterns**), using the above Watered Down Regex, you can exclude the home page. 

To exclude only the home page, you'd enter the following as a URI Exclusion:

```
^/$
```

Where: 

- `^` = beginning of the string
- `$` = end of the string

![screen shot 2016-07-29 at 7 23 27 pm](https://cloud.githubusercontent.com/assets/13220018/17246948/f7aec7b0-55c1-11e6-9d09-f2235f2b57fa.png)

The Watered-down Regular Expression syntax is supported in several existing Comet Cache features and makes it possible to fine-tune the Comet Cache configuration to suit your needs. For more information on the Watered-Down Regular Expression Syntax, see [this KB Article](https://cometcache.com/kb-article/watered-down-regex-syntax/).

**Comment by renzms** [comment]
@raamdev 

Added tags and updated the article, let me know if there's anything else needed, thanks!

**Comment by jaswsinc** [comment]
@renzms I suggest that you also provide a more specific answer to the very specific question that the article title asks, _"How can I exclude the Home Page from being cached?"_.

The response in this article should not only provide the overview and details that you've given, but also an answer that is as specific as the question being asked.

**Comment by raamdev** [comment]
@renzms In addition to Jason's feedback above, note that every KB Article should have more than one paragraph. When a KB Article is published to the live site, the first paragraph gets a larger font than the rest, so it should always provide a brief summary of what the KB Article is about. The following paragraphs should elaborate and provide supporting evidence.

**Comment by raamdev** [comment]
I also suggest a screenshot here, showing the exclusion pattern for excluding the home page from the cache.

**Comment by renzms** [comment]
@jaswsinc @raamdev 

Thanks for the notes guys, I've added a screenshot and some updates above. I'd love to hear your feedback on the update! Thanks! 👍

**Comment by raamdev** [comment]
@renzms Nice! This looks a lot better. Here's some more feedback:

> Many have asked "How can I exclude the Home Page of my website from being cached?", and in this article you will learn how to use URI Exclusion Patterns to do just that.

This needs to be slightly less informal and not just restate what's obvious from the title of the article. See the opening paragraphs on some of the articles [in this list](https://cometcache.com/kb/kb-tag/clearing-the-cache/) for a few ideas.

----

> The solution here is to use the ^ and $ chars in the URI Exclusion Patterns panel. Using Watered-Down Regex, it is possible to set an exclusion that will match at the beginning or end of the URI.

I would swap those sentences so that Watered-Down Regex is mentioned first. Also, I would link to the Watered-Down Regex article in this paragraph instead of the following paragraph—the link should always be used the first time the term is mentioned in an article.

----

> Watered-down Regular Expression syntax is supported in several existing Comet Cache features, including the new list of Custom URLs to Auto-Clear, URI Exclusion Patterns, HTTP Referrer Exclusion Patterns, User-Agent Exclusion Patterns, and the HTML Compressor CSS Exclusion Patterns and JavaScript Exclusion Patterns. This syntax greatly increases the power and flexibility of each of these features and makes things possible like the much-requested ability to Auto-Clear the Home Page or Posts Page of a site whenever a post cache is cleared. For more information on this new watered-down Regular Expression syntax, this KB Article.

All of that doesn't need to be in there. It's not all relevant to the article. I'd shorten this to something like this: "The Watered-down Regular Expression syntax is supported in several existing Comet Cache features and makes it possible to fine-tune the Comet Cache configuration to suit your needs. For more information on the Watered-Down Regular Expression Syntax, [this KB Article](https://cometcache.com/kb-article/watered-down-regex-syntax/)."

**Comment by renzms** [comment]
@raamdev 

Thanks for the tips!

I updated the intro, swapped around a few things like you suggested and simplified the final paragraph. Please see updates above! 👍

**Comment by raamdev** [comment]
@renzms Nice work! I made a few edits in https://github.com/websharks/comet-cache-kb/commit/b120005f1ca1b1678a4040eda7d6957a368a15de and published this. 

Marked as **approved as of 2016-08-15**.
